lib/runners/objc.js

```diff
@@ -105,18 +174,35 @@ module.exports.run = function run(opts, cb) {
             buffer.stderr = '';
             stderr.split(/\n/gm).forEach(line => {
                 let newLine = line.replace(/^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d{3,4} \w*\[[\d:\w]*\]  ?/, '');
+                //remove UnitKit output
+                newLine = newLine.replace(/=== \[[\w \d]*\] ===/, '');
+                //remove StackTrace
+                newLine = newLine.replace(/Stack trace: \(.*\)/, '');
                 if (line == newLine) {
                     buffer.stderr += line + "\n";
                 }
                 else {
-                    buffer.stdout += newLine + "\n";
+                    if (newLine)
+                        buffer.stdout += newLine + "\n";
                 }
             });
 
             // if there is nothing but empty lines, clear the stderr
             if (buffer.stderr.replace(/[ \n]/g, '') == '') {
                 buffer.stderr = '';
             }
+        },
+        sanitizeStdErr: function(error)
+        {
+            error = error || ''
+            return error.replace(/clang.*gnustep-config.*--base-libs.\t*/g, '')
+                        .replace(/Error: Command failed:/g, '')
+                        .replace(/\/home.*(solution\.m|solution)[:0-9]*/g, '')
```

Comment:
this removes the filenames and line number from the error messages because they are basically useless as they are referred to files that the user cannot access. 
The same logic is applied on the C runner